nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -311,6 +319,14 @@ StaticLayer::updateBounds(
   }
 
   std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());
+  processing_layer_.store(true);
+
+  // If there is a new available map, load it.
+  if (map_buffer_ != nullptr) {
```

Comment:
```suggestion
  if (map_buffer_) {
```

the boolean will do what you need